src/server/scripts/EasternKingdoms/ScarletEnclave/chapter1.cpp

```diff
@@ -718,12 +709,17 @@ class npc_dark_rider_of_acherus : public CreatureScript
                 me->SetTarget(TargetGUID);
                 Intro = true;
             }
+			
+			void SpellHitTarget(Unit* target, SpellInfo const* spell) override
+            {
+                if (spell->Id == SPELL_DESPAWN_HORSE && target->GetGUID() == HorseGUID)
+                    if (Creature* creature = target->ToCreature())
+                        creature->DespawnOrUnsummon(2500);
+            }
 
         private:
-            uint32 PhaseTimer;
-            uint32 Phase;
-            bool Intro;
-            ObjectGuid TargetGUID;
```

Comment:
You need to relocate this line instead of removing it.

Reply:
Yeah, sorry my bad because i edited it on another branch to make the pull request! Should be ok now.